### PR TITLE
Don't allow a resumption handshake inside of a SCR

### PR DESCRIPTION
**Pull request by julek-wolfssl**
Fixes ZD16332

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -26394,6 +26408,13 @@ static int HashSkeData(WOLFSSL* ssl, enum wc_HashType hashType,
             return SendTls13ClientHello(ssl);
 #endif
 
+#ifdef HAVE_SECURE_RENEGOTIATION
+        /* We don't want to resume in SCR */
```
Comment (rizlik) [inline src/internal.c:26412]: consider moving this before settings IdSz on line 26404

File: src/internal.c [hunk 2]
```diff
@@ -33618,8 +33642,14 @@ static int DoSessionTicket(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
             if (b == ID_LEN)
                 ssl->options.resuming = 1; /* client wants to resume */
             WOLFSSL_MSG("Client wants to resume session");
-            i += b;
         }
+#ifdef HAVE_SECURE_RENEGOTIATION
+        else {
+            /* We don't want to resume in SCR */
```
Comment (rizlik) [inline src/internal.c:33648]: NIT: consider consolidating this sessionIdSz reset with resuming reset on line 33436.

**Review by rizlik — CHANGES_REQUESTED** [review]
what's the actual need for the` ssl->options.resumed` variable?
It will be true after a renegotiation. If we see a renegotiation as a new handshake, can we still consider the session as resumed?

**Comment by julek-wolfssl** [comment]
retest this please

File: src/internal.c (hunk 2 above)
Comment (julek-wolfssl) [reply src/internal.c:33648]: I'm not sure what you want me to do here?

**Comment by julek-wolfssl** [comment]
Test failures are not related to this PR.

File: src/internal.c (hunk 2 above)
Comment (rizlik) [reply src/internal.c]: If you set `sessionIdSz = 0 ` right after 
```
/* Reset to sane value for SCR */
        ssl->options.resuming = 0;
```
you can avoid this guarded else branch.
It's not blocking for me so marking it as resolved,  feel free to apply it or not

**Review by rizlik — DISMISSED** [review]

**Review by rizlik — APPROVED** [review]

**Review by dgarske — APPROVED** [review]

**Comment by julek-wolfssl** [comment]
Retest this please.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**